src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedStr = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const checker = this[i]
+        ? this[i].toString()
+        : '';
+
+      if (i === this.length - 1) {
+        joinedStr += checker;
+        continue;
+      }
+
+      if (this[i] === null || this[i] === undefined) {
+        joinedStr += checker + separator;
+        continue;
```

Comment:
```suggestion
        joinedStr += checker;
      }

      if (this[i] === null || this[i] === undefined) {
        joinedStr += checker + separator;
```
why do you need `continue` here?

Reply:
Because I have an expression of string concatenation at the end of the loop. Without the continue operator, string concatenation will be done two times in a row with if cases